### PR TITLE
2234 refactor clear simulation data

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Fixes part of #2234 

Refactors ClearSimulationData

Merge after #2276

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running locally and inspecting logs to see that all of the expected functions are called
* [x] If applicable, add screenshots or log transcripts of the feature working

image

### CODE REVIEW

File: monkey/monkey_island/cc/setup/island_event_handlers.py [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
 def setup_island_event_handlers(container: DIContainer):
-    event_queue = container.resolve(PyPubSubIslandEventQueue)
+    island_event_queue = container.resolve(IIslandEventQueue)
+
+    _handle_reset_agent_configuration_events(island_event_queue, container)
+    _handle_clear_simulation_data_events(island_event_queue, container)
 
-    event_queue.subscribe(
+
+def _handle_reset_agent_configuration_events(
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/setup/island_event_handlers.py:17]: `_handle` means that this method will "handle" something. It actually just subscribes to `IslandEventTopic.RESET_AGENT_CONFIGURATION`, the `reset_agent_configuration` is what handles the reset and `island_event_queue` handles the events. I'd suggest to use `_subscribe` prefix instead

File: monkey/tests/unit_tests/monkey_island/cc/resources/test_clear_simulation_data.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+def flask_client(build_flask_client, mock_island_event_queue):
+    container = StubDIContainer()
+    container.register_instance(IIslandEventQueue, mock_island_event_queue)
+
+    with build_flask_client(container) as flask_client:
+        yield flask_client
+
+
+def test_clear_simulation_data(flask_client, mock_island_event_queue):
+    resp = flask_client.post(ClearSimulationData.urls[0], follow_redirects=True)
```
Comment (VakarisZ) [inline monkey/tests/unit_tests/monkey_island/cc/resources/test_clear_simulation_data.py:26]: Why do we need `follow_redirects=True`?

File: vulture_allowlist.py [hunk 3]
```diff
@@ -299,6 +299,9 @@
 deserialize
 serialized_event
 
+# TODO: Remove once #2277 is merged
+clear_simulation_data
```
Comment (VakarisZ) [inline vulture_allowlist.py:303]: Why isn't this imported?

**Review by VakarisZ — APPROVED** [review]
Minor comments/questions

File: vulture_allowlist.py (hunk 3 above)
Comment (ilija-lazoroski) [reply vulture_allowlist.py:303]: It doesn't need to be imported

File: monkey/tests/unit_tests/monkey_island/cc/resources/test_clear_simulation_data.py (hunk 2 above)
Comment (ilija-lazoroski) [reply monkey/tests/unit_tests/monkey_island/cc/resources/test_clear_simulation_data.py:26]: @mssalvatore

Comment (mssalvatore) [reply monkey/tests/unit_tests/monkey_island/cc/resources/test_clear_simulation_data.py:26]: We have it everywhere and I haven't investigated it. For now, consistency. If someone wants to figure out why we use it everywhere, be my guest.

**Pull request merged**